src/main/java/com/globaroman/bookshopproject/service/impl/UserServiceImpl.java

```diff
@@ -1,30 +1,54 @@
 package com.globaroman.bookshopproject.service.impl;
 
+import com.globaroman.bookshopproject.config.PasswordEncoderConfig;
 import com.globaroman.bookshopproject.dto.UserRegistrationRequestDto;
 import com.globaroman.bookshopproject.dto.UserResponseDto;
 import com.globaroman.bookshopproject.exception.RegistrationException;
 import com.globaroman.bookshopproject.mapper.UserMapper;
+import com.globaroman.bookshopproject.model.Role;
+import com.globaroman.bookshopproject.model.User;
+import com.globaroman.bookshopproject.repository.RoleRepository;
 import com.globaroman.bookshopproject.repository.UserRepository;
 import com.globaroman.bookshopproject.service.UserService;
+import java.util.Set;
 import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class UserServiceImpl implements UserService {
+    private static final Long ROLE_ID_IN_DB =
+            (long) Role.RoleName.USER.ordinal() + 1;
     private final UserRepository userRepository;
+    private final RoleRepository roleRepository;
     private final UserMapper userMapper;
+    private final PasswordEncoderConfig passwordEncoderConfig;
```

Comment:
Inject password encoder directly, no need to inject config. 
```suggestion
    private final PasswordEncoderConfig passwordEncoderConfig;
```